tests/integration/security/sds_ingress/ingress_test.go

```diff
@@ -195,7 +195,69 @@ func TestSingleMTLSGateway_CompoundSecretRotation(t *testing.T) {
 			}
 
 			// key/cert rotation
-			ingressutil.RotateSecrets(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialB)
+			ingressutil.RotateSecrets(t, ctx, credName, ingress.Mtls, ingressutil.IngressCredentialB, false)
+			// Use old server CA cert to set up SSL connection would fail.
+			err = ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 0, ErrorMessage: "certificate signed by unknown authority"}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 404 from product page at host %s: %v", host, err)
+			}
+
+			// Use new server CA cert to set up SSL connection.
+			tlsContext = ingressutil.TLSContext{
+				CaCert:     ingressutil.CaCertB,
+				PrivateKey: ingressutil.TLSClientKeyB,
+				Cert:       ingressutil.TLSClientCertB,
+			}
+			err = ingressutil.SendRequest(ing, host, credName[0], ingress.Mtls, tlsContext,
+				ingressutil.ExpectedResponse{ResponseCode: 200, ErrorMessage: ""}, t)
+			if err != nil {
+				t.Fatalf("unable to retrieve 200 from product page at host %s: %v", host, err)
+			}
+		})
+}
+
+// TestSingleMTLSGatewayAndNotGeneric_CompoundSecretRotation tests a single mTLS ingress gateway with SDS enabled and use the tls cert instead of generic cert
+// Verifies behavior in these scenarios.
+// (1) A valid kubernetes secret with key/cert and client CA cert is added, verifies that SSL connection
+// termination is working properly. This secret is a compound secret.
+// (2) After key/cert rotation, client needs to pick new CA cert to complete SSL connection. Old CA
+// cert will cause the SSL connection fail.
+func TestSingleMTLSGatewayAndNotGeneric_CompoundSecretRotation(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			var (
+				credName = []string{"testsinglemtlsgatewaywithcacert-compoundsecretrotation"}
```

Comment:
Please use testsinglemtlsgatewayandnotgeneric-compoundsecretrotation for credName and host.